components/scmprovider.py

```diff
@@ -121,10 +133,9 @@ def check_for_update(self, library, task, ignore_commits_from_these_jobs):
             all_new_upstream_commits = self._commits_since(library.revision)
             if not all_new_upstream_commits:
                 self.logger.log("Checking for updates to %s but no new upstream commits were found from our current in-tree revision %s." % (library.name, library.revision), level=LogLevel.Info)
-                return []
+                return [], []
 
             # We do have new upstream commits.
-            # Step 3: Get the most recent job for the library
             most_recent_job = ignore_commits_from_these_jobs[0] if ignore_commits_from_these_jobs else None
 
             # Step 4: Check if the most recent job was performed on a revision _after_ the library's current revision or _before_.
```

Comment:
Steps in comments need to be renumbered

Reply:
This is also a bug in the main branch I'm going to fix this in a separate commit at the end so i don't need to rebase.